src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ReflectionEmitMemberAccessor.cs

```diff
@@ -294,6 +286,69 @@ private static DynamicMethod CreatePropertySetter(PropertyInfo propertyInfo, Typ
             return dynamicMethod;
         }
 
+        public override Func<object, TProperty> CreateFieldGetter<TProperty>(FieldInfo fieldInfo) =>
+            CreateDelegate<Func<object, TProperty>>(CreateFieldGetter(fieldInfo, typeof(TProperty)));
+
+        private static DynamicMethod CreateFieldGetter(FieldInfo fieldInfo, Type fieldType)
+        {
+            Type? declaringType = fieldInfo.DeclaringType;
+            Debug.Assert(declaringType != null);
+
+            DynamicMethod dynamicMethod = CreateGetterMethod(fieldInfo.Name, fieldType);
+            ILGenerator generator = dynamicMethod.GetILGenerator();
+
+            generator.Emit(OpCodes.Ldarg_0);
+            generator.Emit(
+                declaringType.IsValueType
+                    ? OpCodes.Unbox
+                    : OpCodes.Castclass,
+                declaringType);
+            generator.Emit(OpCodes.Ldfld, fieldInfo);
+            generator.Emit(OpCodes.Ret);
+
+            return dynamicMethod;
+        }
+
+        public override Action<object, TProperty> CreateFieldSetter<TProperty>(FieldInfo fieldInfo) =>
+            CreateDelegate<Action<object, TProperty>>(CreateFieldSetter(fieldInfo, typeof(TProperty)));
+
+        private static DynamicMethod CreateFieldSetter(FieldInfo fieldInfo, Type fieldType)
+        {
+            Type? declaringType = fieldInfo.DeclaringType;
+            Debug.Assert(declaringType != null);
+
+            DynamicMethod dynamicMethod = CreateSetterMethod(fieldInfo.Name, fieldType);
+            ILGenerator generator = dynamicMethod.GetILGenerator();
+
+            generator.Emit(OpCodes.Ldarg_0);
+            generator.Emit(
+                declaringType.IsValueType
+                    ? OpCodes.Unbox
+                    : OpCodes.Castclass,
+                declaringType);
+            generator.Emit(OpCodes.Ldarg_1);
+            generator.Emit(OpCodes.Stfld, fieldInfo);
+            generator.Emit(OpCodes.Ret);
+
+            return dynamicMethod;
+        }
+
+        private static DynamicMethod CreateGetterMethod(string memberName, Type memberType) =>
+            new DynamicMethod(
+                memberName + "Getter",
+                memberType,
+                new[] { JsonClassInfo.ObjectType },
+                typeof(ReflectionEmitMemberAccessor).Module,
+                skipVisibility: true);
+
+        private static DynamicMethod CreateSetterMethod(string memberName, Type memberType) =>
+            new DynamicMethod(
+                memberName + "Setter",
+                typeof(void),
+                new[] { JsonClassInfo.ObjectType, memberType },
+                typeof(ReflectionEmitMemberAccessor).Module,
+                skipVisibility: true);
+
         [return: NotNullIfNotNull("method")]
         private static T? CreateDelegate<T>(DynamicMethod? method) where T : Delegate =>
```

Comment:
Won't work on .NET Framework, will add an extension method to fix the problem for old TFMs, For modern platforms, it won't exist because of conditional compilation.